### PR TITLE
Send Qt key event return in searchfield

### PR DESCRIPTION
**Pull request by ArnaudBienner**
Post a return key event when return is pressed, so the widget behaves like Qt widgets.
Few notes:
- I've added some test code I used, but I'm not sure if you really need it: we playing with the gallery, I guess this can annoying
- I used C++11 override keyword because I'm used to write C++11 code, but it causes a warning (but no compilation error). I can remove it if you want, or you can just keep it in case you plan to migrate to C++11 one day :-)

I've done something similar for Key_Up/Down key events; I will submit it in a separate pull request.

### CODE REVIEW

File: gallery.cpp [hunk 1]
```diff
@@ -73,3 +76,20 @@ Gallery::Gallery(QWidget *parent) : QWidget(parent)
...
+bool Gallery::eventFilter(QObject *watched, QEvent *event)
+{
+    if (watched == searchField) {
+        if (event->type() == QEvent::KeyPress) {
+            QKeyEvent* e = static_cast<QKeyEvent*>(event);
+            switch (e->key()) {
+                case Qt::Key_Return:
+                    QMessageBox::information(this, "Key event received",
+                        "Key event \"" + QKeySequence(e->key()).toString() +
+                        "\" received on searchField");
```
Comment (mikemcquaid) [inline gallery.cpp]: This looks like a debug message; should it be enabled by default?

**Comment by mikemcquaid** [comment]
One query but otherwise :+1:. I'm happy with C++11 being used as long as CI is green :grinning:

File: gallery.cpp (hunk 1 above)
Comment (ArnaudBienner) [inline gallery.cpp]: I wasn't sure how to make visible that my change was working.
Thinking again about this, I think having a simple qDebug() message would be nicer: still visible, but less intrusive.
Or we can simply remove this test code.
What do you think?

Comment (mikemcquaid) [inline gallery.cpp]: Just remove the test-code if that's :cool:, thanks!

**Comment by ArnaudBienner** [comment]
Update done.
Hope it looks OK now :-)

**Comment by mikemcquaid** [comment]
Thanks for the PR :sparkles:!

**Pull request merged**